Makefile

```diff
@@ -142,6 +150,83 @@ calico/felix: bin/calico-felix
 	cp bin/calico-felix docker-image/bin/
 	docker build -t calico/felix docker-image
 
+# Targets for Felix testing with the k8s backend and a k8s API server,
+# with k8s model resources being injected by a separate test client.
+LOCAL_IP_ENV?=$(shell ip route get 8.8.8.8 | head -1 | awk '{print $$7}')
```

Comment:
I'm really keen that we don't run etcd on the host's IP.  It's a real pain for me and it'll clash on jenkins etc.